Modules/getpath.c

```diff
@@ -535,7 +540,11 @@ getpath_realpath(PyObject *Py_UNUSED(self) , PyObject *args)
                 len -= 4;
             }
         }
-        result = PyUnicode_FromWideChar(p, len);
+        if (CompareStringOrdinal(path, (int)pathlen, p, len, TRUE) == CSTR_EQUAL) {
```

Comment:
Can `pathlen` be larger than INT_MAX?

Reply:
No, and if it is, we'll have failed earlier because the API can't handle more than 32K worth of path.